### ISSUE TITLE
Discussion: One contract related to multiple awards

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/standard#790]
A process in Kyrgyz Republic might make independent award decisions for each lot, then group the lots by supplier and sign a single contract for each group of lots.

Award decisions are made at a different time than contracts are signed, and we want award decisions to be disclosed as early as possible. At the time of the award decision, we don't know how the lots will be grouped into contracts. (I assume that, in some cases, a buyer might sign multiple contracts with the same supplier instead of always grouping its lots.) Therefore, the publisher can't disclose the awards such that they can avoid (in this case) a one-to-many mapping from contract to awards.

I don't know an elegant solution besides having e.g. `awardIDs` on `Contract`.

Additional note: We should clarify in the documentation that the "Award" means the "Award decision".

Related: #249

**Comment by JachymHercher** [comment]
In eForms, a lot can receive one or more bids and a contract can be linked to one or more winning bids, which I think is equivalent to the situation described above. This seems to be a common use case for example in the medical sector, where a procedure can contain hundreds of lots for particular pharmaceuticals / chemical substances and a few pharma companies submit their bids for all lots. Once the bids are evaluated, a contract is signed with each bidder, not per lot, as that would be very cumbersome.

**Comment by yolile** [comment]
Could this be modeled as: create an award for each supplier, but without declaring `award/items`. Then, for each supplier, create a contract that relates to its award and disclose the items in `contract/items`?

**Comment by jpmckinney** [comment]
To narrow the issue:

* a lot can receive one or more bids
  * Covered by [bids.details.relatedLots](https://extensions.open-contracting.org/en/extensions/lots/master/schema/#bid)
* a contract can be linked to one or more winning bids
  * Presently covered for a single bid by [awards.relatedBid](relatedBid) (to which a contract is related via `contracts.awardID`). There is an issue to [make awards.relatedBid an array](https://github.com/open-contracting/ocds-extensions/issues/125).
* a contract is signed with each bidder, not per lot
  * Covered by [awards.relatedLots](https://extensions.open-contracting.org/en/extensions/lots/master/schema/#award), to link multiple lots to one award

In the issue description, the scenario (which we might want to validate) is that apparently the Kyrgyz Republic *does* do the cumbersome step of awarding per lot, and only later signs contracts per bidder.

This now strikes me as unusual. Signing per lot is certainly more paperwork than awarding per lot since it involves an external party (the bidder) who needs to review and sign each contract – but it still seems like a lot of overhead to award per lot...

This scenario might be a misrepresentation of reality, e.g. if the scenario that was reported to us was constructed based on data in a database – rather than on information as announced in notices, etc. or on descriptions in legal instruments.

**Comment by jpmckinney** [comment]
> Could this be modeled as: create an award for each supplier, but without declaring `award/items`. Then, for each supplier, create a contract that relates to its award and disclose the items in `contract/items`?

I don't think so, if the goal is to provide as much information as early as possible. In the scenario described, the items per supplier are known (in separate awards), it's just that they are eventually bundled into a single contract.

**Comment by JachymHercher** [comment]
I'll try to summarize, just to see whether we are on the same page: 
- We need to validate the scenario.
- If the scenario is validated, then the only solution we have is turning `awardID` into an array, which shouldn't cause any problems (?) but it's a backwards incompatible change, so it has to wait for 2.0.

**Comment by jpmckinney** [comment]
> If the scenario is validated, then the only solution we have is turning awardID into an array, which shouldn't cause any problems (?) but it's a backwards incompatible change, so it has to wait for 2.0.

The other solution is to recommend postponing the disclosure of items, from https://github.com/open-contracting/standard/issues/790#issuecomment-718801713

Since I'm not sure the problem actually exists, and since the issue was reported via an implementer who might have based their question on the structure of the database, I think okay to close until we have further evidence.

At any rate, I'm okay with postponing the disclosure of items, since in practice that is what other buyers do (they only announce the final "bundles" of items per supplier – not any intermediate decisions about each lot).

**Comment by jpmckinney** [comment]
Buyers can award multiple lots to a single supplier on different dates, and then sign a single contract. From https://github.com/open-contracting/standard/pull/1175#discussion_r628153740

> For example, an award for one of the lots may be delayed compared to the other ones because the company is answering clarification questions from the buyer about its offer, but they want to move forward with the other lots.

In OCDS 1.1, we can't have multiple award objects referenced by a single contract object, which:

1. makes it difficult to implement the above scenario (I suppose the publisher updates an existing award object if an additional lot is awarded to the same supplier)
2. makes it hard to model the date and value of the awarded lot (does this go on the lot object, which appears only under `tender.lots`? https://github.com/open-contracting/ocds-extensions/issues/162).

Related: https://github.com/open-contracting/standard/issues/891#issuecomment-832277089 copied below:

> There are deficiencies in the lots modelling, that interacts with deficiencies in the awards/contracts modelling.
> 
> A contract object in OCDS should match as much as possible the real-world contract, so that we have the same number of contracts in OCDS as in the real world, with the same values, etc.
> 
> Now, multiple lots with distinct values can be awarded on different dates to the same supplier, and later combined into one contract. However, a contract has a single awardID; as such, the lots' awards need to be combined into a single award. (The alternative is to have one award per lot, each with one contract, but then this breaks the above expectation.) When combining the awards, we lose the information about each lot's value and awarded date. We instead have the total value of all lots and _some_ date.
> 
> I think changing the cardinality of awardID is too significant a change for a minor release. The only options then are to accept the information loss, or to change the lots extension ([open-contracting/ocds-extensions#162](https://github.com/open-contracting/ocds-extensions/issues/162)). However, this would create very unusual and hard-to-describe semantics (viz. #895 #1175), e.g. an award is as currently defined in #895, except if multiple lots are awarded to the same supplier, in which case an award is an aggregate of other awards...
>
> Originated in https://github.com/open-contracting/standard/pull/1175#discussion_r624672836

**Comment by jpmckinney** [comment]
We don't have any promising proposed solutions, and there is no clear demand, so rescheduling for 1.3 (or guidance in the meantime).
